state-chain/pallets/cf-pools/src/lib.rs

```diff
@@ -1523,26 +1524,81 @@ impl<T: Config> Pallet<T> {
 		Ok(assets_change)
 	}
 
+	pub fn try_add_limit_orders(
+		account_id: &T::AccountId,
+		from: any::Asset,
+		to: any::Asset,
+		maybe_limit_orders: Option<Vec<(Tick, U256)>>,
+	) -> Result<(), DispatchError> {
+		let Some(limit_orders) = maybe_limit_orders else {
+			return Ok(());
+		};
+
+		let (asset_pair, side) = AssetPair::from_swap(from, to).ok_or("Invalid asset pair")?;
+
+		Self::try_mutate_pool(asset_pair, |_, pool| {
+			for (id, (tick, amount)) in limit_orders.into_iter().enumerate() {
+				let order_id = (account_id.clone(), id as u64);
+				pool.pool_state
+					.collect_and_mint_limit_order(&order_id, !side, tick, amount)
+					.map_err(|_| "Failed to set limit order")?;
+			}
+
+			Ok(())
+		})
+	}
+
+	#[allow(clippy::type_complexity)]
 	#[transactional]
 	pub fn swap_with_network_fee(
 		from: any::Asset,
 		to: any::Asset,
 		input_amount: AssetAmount,
+		additional_limit_orders: Option<(
+			T::AccountId,
+			Option<Vec<(Tick, U256)>>,
```

Comment:
what's the rationale behind having these be options inside here? vs. having just the outer option, and then having these just be an empty Vec  when there's nothing

Reply:
If we want to early exit, we can always do `is_empty()`